addons/website/static/src/js/editor/snippets.options.js

```diff
@@ -932,7 +932,12 @@ options.Class.include({
     _select: async function (previewMode, widget) {
         await this._super(...arguments);
 
-        if (this.options.isWebsite && !widget.$el.closest('[data-no-widget-refresh="true"]').length) {
+        // Some blocks flicker when we start their public widgets, so we skip
+        // the refresh for them to avoid the flickering.
+        const targetNoRefreshSelector = ".s_instagram_page";
+
+        if (this.options.isWebsite && !widget.$el.closest('[data-no-widget-refresh="true"]').length
+            && !this.$target[0].matches(targetNoRefreshSelector)) {
```

Comment:
Not a bad idea for now but add a TODO comment to say we should review the way public widget are restarted when converting to OWL and a new API.